### PR TITLE
Suggested changes to #46

### PR DESCRIPTION
**Pull request by droyad**

### CODE REVIEW

File: source/Octostache/Templates/AnalysisContext.cs [hunk 1]
```diff
@@ -6,11 +6,9 @@ namespace Octostache.Templates
 {
     class AnalysisContext
     {
-        readonly AnalysisContext? parent;
-        readonly Identifier? identifier;
-        readonly SymbolExpression? expansion;
-
-        public AnalysisContext() {}
```
Comment (droyad) [inline source/Octostache/Templates/AnalysisContext.cs:13]: Caller of this wasn't used

File: source/Octostache/Templates/BuiltInFunctions.cs [hunk 2]
```diff
@@ -33,15 +33,6 @@ static class BuiltInFunctions
             {"uripart", TextManipulationFunction.UriPart}
         };
 
-        // Configuration should be done at startup, this isn't thread-safe.
-        public static void Register(string name, Func<string?, string[], string?> implementation)
```
Comment (droyad) [inline source/Octostache/Templates/BuiltInFunctions.cs:37]: Dead

File: source/Octostache/Templates/EvaluationContext.cs [hunk 3]
```diff
@@ -14,14 +14,14 @@ class EvaluationContext
         readonly EvaluationContext? parent;
         readonly Stack<SymbolExpression> symbolStack = new Stack<SymbolExpression>();
 
-        public EvaluationContext(Binding binding, TextWriter? output, EvaluationContext? parent = null)
```
Comment (droyad) [inline source/Octostache/Templates/EvaluationContext.cs:17]: Passed `TextWriter.Null` instead

File: source/Octostache/Templates/TemplateAnalyzer.cs [hunk 4]
```diff
@@ -6,12 +6,7 @@ namespace Octostache.Templates
 {
     static class TemplateAnalyzer
     {
-        public static IEnumerable<string> GetDependencies(Template template)
```
Comment (droyad) [inline source/Octostache/Templates/TemplateAnalyzer.cs:9]: Dead code

File: source/Octostache/Templates/TemplateParser.cs [hunk 5]
```diff
@@ -52,7 +52,7 @@ public static class TemplateParser
         static readonly Parser<Indexer> SymbolIndexer =
             (from index in Substitution.Token()
              where index.Expression is SymbolExpression
-             select new Indexer((index.Expression as SymbolExpression)!))
+             select new Indexer((SymbolExpression) index.Expression))
```
Comment (slewis74) [inline source/Octostache/Templates/TemplateParser.cs:55]: Nice, didn't think of this at the time.

**Review by slewis74 — APPROVED** [review]
Looks good 👍

**Pull request merged**